array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,39 @@
+/* exported chunk */
+// set condition for the empty array (size > array.length) and return array;
+// set condition for (size === 1) and run traditional for loop
+// array.slice(i, i+size) for ex (0, 1), (1, 2), (2, 3) etc. to break up into single chunks and push
+// set condition for (array.length === 7) and slice it up 3 ways (0, size), (size, 6), and (-1);
+// unshift sliceOne, push sliceTwo, and push sliceThree;
+// for the rest, repeat process for (array.length === 7), but just slice it twice
+function chunk(array, size) {
+  if (size > array.length) {
+    return array;
+  }
+  if (size === 1) {
+    var newResult = [];
+    for (var i = 0; i < array.length; i++) {
+      var oneSlice = array.slice(i, i + size);
+      newResult.push(oneSlice);
+    }
+    return newResult;
+  }
+  if (array.length === 7) {
```

Comment:
Why specifically seven? This seems like you have a bug in your process that you are trying to cover up